### PR TITLE
Upgraded nextjs deps and configuration

### PR DESCRIPTION
**Pull request by pnicolli**

### CODE REVIEW

**Comment by sneridagh** [comment]
@pnicolli I am one of the late "ESM" all the things. How about declare the nextjs app `type: module`?

**Comment by pnicolli** [comment]
@sneridagh I didn't dig deep into that world yet. What would be the change you are proposing? What I did here is basically to create an empty nextjs project with the suggested tooling in the nextjs docs and copy the config over here 😛

**Comment by sneridagh** [comment]
@pnicolli yeah, basically then you don't have to do the `.mjs` dance anymore. I don't think there are other implications from the app point of view, unless you have to deal with importing .cjs files around.

**Comment by pnicolli** [comment]
@sneridagh I gave it some thought and I think `type: module` is not a good fit for this app. I would like to keep it as close as possible to out-of-the-box nextjs.

I merged main and fixed a couple eslint warnings, the PR about providers would also benefit from this PR imho.

File: apps/nextjs/.gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+.yarn/install-state.gz
```
Comment (sneridagh) [inline apps/nextjs/.gitignore:7]: ?

**Review by sneridagh — APPROVED** [review]
LGTM, this can go before the providers refactor

File: apps/nextjs/.gitignore (hunk 1 above)
Comment (pnicolli) [reply apps/nextjs/.gitignore:7]: It's in the latest create-next-app basic template 😅

**Pull request merged**